In [3]:
import pandas as pd 

health = pd.read_csv('healthcare_dataset.csv')
health

Name  Age  Gender Blood Type Medical Condition  \
0          Bobby JacksOn   30    Male         B-            Cancer   
1           LesLie TErRy   62    Male         A+           Obesity   
2            DaNnY sMitH   76  Female         A-           Obesity   
3           andrEw waTtS   28  Female         O+          Diabetes   
4          adrIENNE bEll   43  Female        AB+            Cancer   
...                  ...  ...     ...        ...               ...   
55495  eLIZABeTH jaCkSOn   42  Female         O+            Asthma   
55496         KYle pEREz   61  Female        AB-           Obesity   
55497       HEATher WaNG   38  Female         B+      Hypertension   
55498     JENniFER JOneS   43    Male         O-         Arthritis   
55499       jAMES GARCiA   53  Female         O+         Arthritis   

      Date of Admission            Doctor                      Hospital  \
0            2024-01-31     Matthew Smith               Sons and Miller   
1            2019-08-20   Samantha Davies                       Kim Inc   
2            2022-09-22  Tiffany Mitchell                      Cook PLC   
3            2020-11-18       Kevin Wells    Hernandez Rogers and Vang,   
4            2022-09-19    Kathleen Hanna                   White-White   
...                 ...               ...                           ...   
55495        2020-08-16     Joshua Jarvis                Jones-Thompson   
55496        2020-01-23   Taylor Sullivan                  Tucker-Moyer   
55497        2020-07-13    Joe Jacobs DVM  and Mahoney Johnson Vasquez,   
55498        2019-05-25    Kimberly Curry      Jackson Todd and Castro,   
55499        2024-04-02     Dennis Warren                Henry Sons and   

      Insurance Provider  Billing Amount  Room Number Admission Type  \
0             Blue Cross    18856.281306          328         Urgent   
1               Medicare    33643.327287          265      Emergency   
2                  Aetna    27955.096079          205      Emergency   
3               Medicare    37909.782410          450       Elective   
4                  Aetna    14238.317814          458         Urgent   
...                  ...             ...          ...            ...   
55495         Blue Cross     2650.714952          417       Elective   
55496              Cigna    31457.797307          316       Elective   
55497   UnitedHealthcare    27620.764717          347         Urgent   
55498           Medicare    32451.092358          321       Elective   
55499              Aetna     4010.134172          448         Urgent   

      Discharge Date   Medication  Test Results  
0         2024-02-02  Paracetamol        Normal  
1         2019-08-26    Ibuprofen  Inconclusive  
2         2022-10-07      Aspirin        Normal  
3         2020-12-18    Ibuprofen      Abnormal  
4         2022-10-09   Penicillin      Abnormal  
...              ...          ...           ...  
55495     2020-09-15   Penicillin      Abnormal  
55496     2020-02-01      Aspirin        Normal  
55497     2020-08-10    Ibuprofen      Abnormal  
55498     2019-05-31    Ibuprofen      Abnormal  
55499     2024-04-29    Ibuprofen      Abnormal  

[55500 rows x 15 columns]

In [10]:
health.describe()

Age  Billing Amount   Room Number
count  55500.000000    55500.000000  55500.000000
mean      51.539459    25539.316097    301.134829
std       19.602454    14211.454431    115.243069
min       13.000000    -2008.492140    101.000000
25%       35.000000    13241.224652    202.000000
50%       52.000000    25538.069376    302.000000
75%       68.000000    37820.508436    401.000000
max       89.000000    52764.276736    500.000000

# QN 1

Explain the central tendency and distribution of two continuous variables

In [12]:
# for Billing Amount and age
# median

billing_amount_median = health['Billing Amount'].median()
age_median = health['Age'].median()
print({'Billing Amount': billing_amount_median, 'Age': age_median})

{'Billing Amount': np.float64(25538.069375965664), 'Age': np.float64(52.0)}


In [13]:
# mean
billing_amount_mean = health['Billing Amount'].mean()
age_mean = health['Age'].mean()
print({'Billing Amount': billing_amount_mean, 'Age': age_mean})

{'Billing Amount': np.float64(25539.316097211795), 'Age': np.float64(51.53945945945946)}


In [14]:
# distributio/dispersion
# variance
billing_amount_var = health['Billing Amount'].var()
age_var = health['Age'].var()
print({'Billing Amount': billing_amount_var, 'Age': age_var})

{'Billing Amount': np.float64(201965437.04053578), 'Age': np.float64(384.2561953149387)}


In [15]:
# standard deviation
billing_amount_std = health['Billing Amount'].std()
age_std = health['Age'].std()
print({'Billing Amount': billing_amount_std, 'Age': age_std})

{'Billing Amount': np.float64(14211.454430864414), 'Age': np.float64(19.602453808514348)}


            Explanation
For Billing amount, the mean is: 25539.316097211795, and the median is: 25538.069375965664
For Age the mean is: 51.53945945945946, the median is: 52.0
For Billing Amount the variance is: 201965437.04053578, and the standard deviation is:14211.454430864414
For Age the variance is:384.2561953149387 ,and the standard deviation is:19.602453808514348 

# QN .2
Explain the marginal probability of any two categorical variables

In [6]:
# finding the marginal probability of gender
# let F represent FEMALES

event_F = len(health[health['Gender'] == 'Female'])
probability_of_F = event_F/len(health)
print('the probability of females is: ', probability_of_F)

the probability of females is:  0.4995675675675676


In [7]:
#let E represent Emergency admission
event_E = len(health[health['Admission Type'] == 'Emergency'])
probability_of_E = event_E/len(health)
print('the probability of getting an emergency is: ', probability_of_E)

the probability of getting an emergency is:  0.3291711711711712


            Explanation
The marginal probability of females is 0.49957 approximately 0.5 
The marginal probability of getting an emergency type of admission at all hopspitals is 0.329

# QN 3 

Explain the relationship between 'medical condition' and 'Admission Type'

In [8]:
import scipy.stats 
from scipy.stats import chi2_contingency

In [9]:
# performing a chi_square test to get the relationship
# creating a contigency matrix 
medicalCondition_AdmissionType = pd.crosstab(index= health['Medical Condition'], columns= health['Admission Type'] )

# the chi square test
chi2_stat, p, dof, expected = scipy.stats.chi2_contingency(medicalCondition_AdmissionType)
print(f'chi_statistics {chi2_stat:.5g}')
print(f'p value  {p:.5g}')
print(f'degree of freedom {dof}')
print('The expected value is:\n', expected)

chi_statistics 19.338
p value  0.036172
degree of freedom 10
The expected value is:
 [[3128.66198198 3063.92526126 3115.41275676]
 [3087.31846847 3023.43720721 3074.24432432]
 [3101.43576577 3037.2623964  3088.30183784]
 [3127.31747748 3062.60857658 3114.07394595]
 [3107.48603604 3043.18747748 3094.32648649]
 [3102.78027027 3038.57908108 3089.64064865]]


            Explanation
The chi_stat value ranges from 0 to infinity.., in this case 